resources/src/operation/mod.rs

```diff
@@ -0,0 +1,220 @@
+                    CreateAccountFields::new(rep.account, rep.funder, rep.starting_balance)
+                        .map_err(|err| de::Error::custom(err))?;
+                Ok(Operation {
+                    id: rep.id,
+                    paging_token: rep.paging_token,
+                    type_i: rep.type_i,
+                    operation_detail: OperationDetail::CreateAccount(operation_detail),
+                })
+            }
+            _ => Err(de::Error::custom("Invalid operation type.")),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PaymentFields {
+    from: String,
+    to: String,
+    asset: AssetIdentifier,
+    amount: Amount,
+}
+
+/// A path payment operation represents a payment from one account to another through a path. This
+/// type of payment starts as one type of asset and ends as another type of asset. There can be
+/// other assets that are traded into and out of along the path.
+#[derive(Debug, Deserialize, Serialize)]
+struct PathPayment {
+    from: String,
+    to: String,
+    destination_asset: AssetIdentifier,
+    destination_amount: Amount,
+    source_asset: AssetIdentifier,
+    source_max: Amount,
+    source_amount: Amount,
+}
+
+/// A “Manage Offer” operation can create, update or delete an offer to trade assets in the Stellar
+/// network. It specifies an issuer, a price and amount of a given asset to buy or sell.
+#[derive(Debug, Deserialize, Serialize)]
+struct ManageOffer {
+    offer_id: i64,
+    selling: AssetIdentifier,
+    buying: AssetIdentifier,
+    amount: Amount,
+    #[serde(rename = "price_r")] price_ratio: PriceRatio,
+    price: Amount,
+}
+
+/// “Create Passive Offer” operation creates an offer that won’t consume a counter offer that
+/// exactly matches this offer. This is useful for offers just used as 1:1 exchanges for path
+/// payments. Use Manage Offer to manage this offer after using this operation to create it.
+#[derive(Debug, Deserialize, Serialize)]
+struct CreatePassiveOffer {
+    offer_id: i64,
+    selling: AssetIdentifier,
+    buying: AssetIdentifier,
+    amount: Amount,
+    #[serde(rename = "price_r")] price_ratio: PriceRatio,
+    price: Amount,
+}
+
+/// Use “Set Options” operation to set following options to your account:
+///
+/// Set/clear account flags:
+/// AUTH_REQUIRED_FLAG (0x1) - if set, TrustLines are created with authorized set to false
+/// requiring the issuer to set it for each TrustLine.
+/// AUTH_REVOCABLE_FLAG (0x2) - if set, the authorized flag in TrustLines can be cleared.
+/// Otherwise, authorization cannot be revoked.
+/// Set the account’s inflation destination.
+/// Add new signers to the account.
+/// Set home domain.
+#[derive(Debug, Deserialize, Serialize)]
+struct SetOptions {
+    signer_key: String,
+    signer_weight: u8,
+    master_key_weight: u8,
+    low_threshold: u32,
+    med_threshold: u32,
+    high_threshold: u32,
+    home_domain: String,
+    set_flags: Option<Vec<u32>>,
```

Comment:
Does this need both sets of flag arrays for the internal type? Also, can we map the flags to an enum? Possibly even do the same with account if we haven't already?